ambry-frontend/src/main/java/com/github/ambry/frontend/AmbrySecurityService.java

```diff
@@ -125,6 +129,12 @@ public void postProcessRequest(RestRequest restRequest, Callback<Void> callback)
         exception = new RestServiceException("Too many requests", RestServiceErrorCode.TooManyRequests);
       } else if (storageQuotaService != null && storageQuotaService.shouldThrottle(restRequest)) {
         exception = new RestServiceException("StorageQuotaExceeded", RestServiceErrorCode.TooManyRequests);
+      } else if (quotaManager != null && quotaManager.getQuotaConfig().requestThrottlingEnabled) {
```

Comment:
This is not right, this `else if` block should be moved to `else` block down below. If `quotaManager is not null and request throttling is enabled`, then the execution flow would go inside this`else if` block and never go to the `else` block down below. This is not what we want to, we should change it like this
```
} else {
    if (quotaManager != null && quotaManager.getQuotaConfig().requestThrottlingEnabled) {
                ThrottlingRecommendation throttlingRecommendation = quotaManager.getThrottleRecommendation(restRequest);
        if (throttlingRecommendation != null && throttlingRecommendation.shouldThrottle()) {
          // TODO Use ThrottlingRecommendation to set response headers about quota usage.
          throw new RestServiceException("User Quota Exceeded", RestServiceErrorCode.TooManyRequests);
        }
    }

    if (restRequest.getRestMethod() == RestMethod.DELETE || ....)
}
```